src/main/java/com/doit/api/Item/ItemService.java

```diff
@@ -0,0 +1,47 @@
+package com.doit.api.Item;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ItemService {
+
+    private final ItemRepository itemRepository;
+
+    @Autowired
+    public ItemService(ItemRepository itemRepository) {
+        this.itemRepository = itemRepository;
+    }
+
+    public String addItem(ItemRequest request) {
+        Item item = Item.builder()
+                .name(request.getName())
+                .itemState(request.getItemState())
+                .build();
+        boolean itemExists = itemRepository.findByName(item.getName()).isPresent();
+        if (item.getName().length() > 100)
+            return "TOO LARGE";
+        else if (itemExists)
+            return ("CONFLICT");
+        else {
+            itemRepository.save(item);
+            return "ADDED";
+        }
```

Comment:
We can move those checks to the `ItemController` and in that case, we wouldn't need to return those strings. Instead, we can return just a boolean.